src/noderefs.rs

```diff
@@ -0,0 +1,109 @@
+use rustc_hash::FxHashSet;
+use std::hash::Hash;
+use std::iter::FromIterator;
+use std::ops::Deref;
+
+/// A set of node references.
+///
+/// Can be created from a single node reference or an iterable of
+/// node references.
+///
+/// ```
+/// use pathfinding::NodeRefs;
+///
+/// #[derive(Debug, Clone, PartialEq, Eq, Hash)]
+/// struct N(String);
+///
+/// let red = N("red".into());
+/// let blue = N("blue".into());
+/// let green = N("green".into());
+///
+/// let refs: NodeRefs<N> = NodeRefs::from(&red);
+/// let refs: NodeRefs<N> = NodeRefs::from_iter([&red, &blue, &green]);
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct NodeRefs<'a, N>(FxHashSet<&'a N>)
```

Comment:
I decided that this struct doesn't need to own the values - similar to how we pass `&N` as start currently for most algorithms instead of `N`.